docs/adr/0005-adding-line-items.md

```diff
@@ -0,0 +1,29 @@
+# 4. Notification error handling.
+
+Date: 2022-02-01
+
+## Status
+
+[On Review]()
+
+## Context
+
+- Many payment methods requires information about the price and product information by specifying `lineItems` in `makePaymentRequest`.
+- Integration supports automatically **adding lineItems** from commercetools cart for [klarna and affirm payments](https://github.com/commercetools/commercetools-adyen-integration/blob/v9.6.0/extension/docs/WebComponentsIntegrationGuide.md#klarna-payment-and-affirm-payment).
+- There are many payment methods which require line items **beside** the klarna and affirm payment.
+- Since we add out of the box for Affirm and Klarna only the users need to write additional logic to include those items for other payment methods which poses an inconvenience. For instance check the request here on issue [#728](https://github.com/commercetools/commercetools-adyen-integration/issues/728).
+
+## Decision
+
+- If lineItems field is provided with `makePaymentRequest`, extension will skip adding lineItems and leave the provided lineItems.
+- An application configuration flag `addCommercetoolsLineItems: true/false` for extension, which will default to false and if set to true would add lineItems to all payment methods.
+- If user includes in the makePaymentRequest,  "addCommercetoolsLineItems": true then integration would generate line items, no matter if it's set to false on application config.
```

Comment:
is it something like forcing line item generation for other payment methods like credit card ? 🤔

Reply:
So this point: 
```
Extension does not include lineItems for payment methods where those infos are not beneficial and thus we save an API call i.e: fetch the cart.
```
Will it work if the payment method is not the one that requires [lineItems]?

Reply:
Yes it works, but not required. I guess those add additional fields on response related to fraud calculation, so does not seem harmful.

https://github.com/commercetools/commercetools-adyen-integration/pull/918/commits/bae78ed26e44a24a3c49de3412182d4358ec40c2#diff-e2e07d6c2da8d9a9bdf7f434ef5c8c01b699a20c81feb1d379dbc12317afd5caR44-R48